src/routes/relay/relay.controller.spec.ts

```diff
@@ -49,151 +65,298 @@ describe('RelayController', () => {
   });
 
   describe('/v1/relay (POST)', () => {
-    it('should return a 201 when the body is valid', async () => {
-      axios.default.get = jest.fn().mockResolvedValue({ data: 'mockSafe' });
-
-      const relayServiceSpy = jest
-        .spyOn(relayService, 'sponsoredCall')
-        .mockImplementation(() => Promise.resolve({ taskId: '123' }));
-
-      const body = {
-        chainId: '5',
-        to: faker.finance.ethereumAddress(),
-        data: MOCK_EXEC_TX_CALL_DATA,
-      };
-
-      await request(app.getHttpServer())
-        .post('/v1/relay')
-        .send(body)
-        .expect(201);
-
-      expect(relayServiceSpy).toHaveBeenCalledTimes(1);
+    describe('Relayer', () => {
+      it('should return a 201 when the body is valid', async () => {
+        axios.default.get = jest.fn().mockResolvedValue({ data: 'mockSafe' });
+
+        const sponsorServiceSpy = jest
+          .spyOn(mockSponsorService, 'sponsoredCall')
+          .mockImplementation(() => Promise.resolve({ taskId: '123' }));
+
+        const body = {
+          chainId: '5',
+          to: faker.finance.ethereumAddress(),
+          data: MOCK_EXEC_TX_CALL_DATA,
+        };
+
+        const result = await request(app.getHttpServer())
+          .post('/v1/relay')
+          .send(body)
+          .expect(201);
+
+        expect(result.body).toStrictEqual({ taskId: '123' });
+
+        expect(sponsorServiceSpy).toHaveBeenCalledTimes(1);
+      });
+
+      it('should return a 500 if the relayer throws', async () => {
+        axios.default.get = jest.fn().mockResolvedValue({ data: 'mockSafe' });
```

Comment:
`axios` should also be encapsulated somehow. Maybe you already have plans for it but it'll make these tests less dependent on the implementation and easier to read.

Reply:
We will indeed tackle this next.